.travis.yml

```diff
@@ -4,58 +4,38 @@ branches:
-    # Currently, Travis allows a maximum of two concurrent Mac builds per owner:
-    # https://blog.travis-ci.com/2017-09-22-macos-update
-    # With the following order, xcode7.3 will take up the first slot, and at the same time the three other
-    # environments use the second slot in sequence. xcode10.1 will finish first and provide some early results - most
-    # test failures happen in all environments and will be quickly reported this way.
-
     # OSX 10.11 El Capitan
     - osx_image: xcode7.3
+    # macOS 10.12 Sierra
+    - osx_image: xcode9.2
+    - osx_image: xcode8.3
     # macOS 10.13 High Sierra
     - osx_image: xcode10.1
     # macOS 10.14 Mojave
     - osx_image: xcode10.2
-    # macOS 10.12 Sierra
-    - osx_image: xcode9.2
-    - osx_image: xcode8.3
 language: generic
 cache:
   directories:
     ~/.pyenv
-env:
-  global:
-    - PY35="3.5.6"
-    - PY36="3.6.8"
-    - PY37="3.7.2"
 install:
-  # The different Travis environments have different pyenv versions, and not all have the Python versions we need,
-  # so we upgrade pyenv to get a consistent setup in all environments.
-  - brew update
-  - brew outdated pyenv || brew upgrade pyenv
-  # The existing pyenv shims from the build cache may no longer be valid, for example if pyenv was upgraded
-  # (in which case the paths used in the shims point to an older, no longer installed pyenv version).
-  # By rehashing the shims we can be sure that they are always valid again.
-  - pyenv rehash
   # pyenv's shims directory is not in the PATH by default.
   - export PATH="$(pyenv root)/shims:${PATH}"
-  # These are the Python versions that we want to test on.
-  - pyenv install --skip-existing ${PY35}
-  - pyenv install --skip-existing ${PY36}
-  - pyenv install --skip-existing ${PY37}
-  - pyenv global ${PY37} ${PY36} ${PY35} system
-  - pip install --upgrade tox
-  - make -f Makefile
+  # Diagnostic: Find out which versions of Python are availalbe.
+  - pyenv install -l
+  # Rubicon supports Python 3.5 and newer.
+  # The most recent version of Python 3.5 provided by the pyenv install on
+  # macOS 10.11 is 3.5.2. There are more recent versions of Python 3.5
+  # available, but using them would mean updating pyenv, which takes 10+
+  # minutes on macOS 10.11.
+  - pyenv install --skip-existing 3.5.2
+  - pyenv global 3.5.2
+  - python -m pip install --upgrade pip setuptools
+  - python -m pip install .
 script:
-  - export DYLD_LIBRARY_PATH=`pwd`/tests/objc
-  - tox -e "py{35,36,37},flake8"
+  - make -f Makefile
+  # This next line *should* be:
+  #   DYLD_LIBRARY_PATH=tests/objc python setup.py test
+  # but Travis futzes with DYLD_LIBRARY_PATH, so that doesn't work.
+  # This sets the environment variable in such a way that Travis can't
+  # intercept it at the shell level.
+  - python -c "import os, subprocess; env=os.environ.copy(); env['DYLD_LIBRARY_PATH']='tests/objc'; subprocess.run('python setup.py test'.split(), env=env)"
```

Comment:
See https://github.com/beeware/rubicon-objc/pull/145#issuecomment-570780531 - the explanation is not entirely correct (it's macOS that messes with the `DYLD_LIBRARY_PATH` variable, not Travis), and I'm *relatively* sure that there should be a way to `DYLD_LIBRARY_PATH` without this workaround.

Reply:
There has to be *something* Travis specific, because the simple version works without any difficulties on GitHub Actions. But your comment about full paths and/or system python confusion gives another couple of possible avenues for a simpler approach. I'll keep poking.

Reply:
Solved below: https://github.com/beeware/rubicon-objc/pull/145#discussion_r363059789